### ISSUE TITLE
 False positive from clang-analyzer-deadcode.DeadStores with std::ref and std::bind

### ISSUE THREAD

**Issue by randombit** [llvm/llvm-project#53597]
```
#include <functional>
#include <stdio.h>

void run(int v)
   {
   printf("%d\n", v);
   }

int main()
   {
   int v = 5;
   auto f = std::bind(run, std::ref(v));
   f();

   v = 10;
   f();
   }
```

The `printf` in `run` is just to "prove" that the second store is used.

```
1 warning generated.
/home/jack/ref.cpp:15:4: warning: Value stored to 'v' is never read [clang-analyzer-deadcode.DeadStores]
   v = 10;
   ^   ~~
/home/jack/ref.cpp:15:4: note: Value stored to 'v' is never read
   v = 10;
   ^   ~~
```

```
$ clang-tidy --version
LLVM (http://llvm.org/):
  LLVM version 13.0.0
  Optimized build.
  Default target: x86_64-pc-linux-gnu
  Host CPU: skylake
```

**Comment by CaseyCarter** [comment]
Sorry: closed inadvertently by typo.
